### ISSUE TITLE
Cannot use plugin after a chat with a non-existent name was joined.

### ISSUE THREAD

**Issue by michaelzangl** [jgeboski/purple-facebook#104]
What I did:
- Add an existing, group chat on Facebook to the buddy list
- In the following dialog, enter something other than that number in "chat name".
- Double-click on it to join.
- Click on OK on the following join dialog.

The connection was cancelled instantly. The chat thus disappeared from the list. I had to edit the blist.xml and disable autojoining to be able to use Facebook again.

I would expect the plugin to be able to handle such simple user errors (like giving the chat a meaningful name when they are asked for the "Chat Name").

```
(18:34:34) g_log: purple_conversation_get_chat_data: assertion `conv != NULL' failed
(18:34:34) http: Performing new request 0x7f3933599d10 for https://graph.facebook.com/graphql.
(18:34:34) http: Connecting to https://graph.facebook.com/graphql...
(18:34:34) facebook: HTTP Request (0x7f3933599d10):
(18:34:34) facebook:   Request URL: https://graph.facebook.com/graphql
(18:34:34) facebook:   Request Data: fb_api_req_friendly_name=ThreadQuery&locale=de_DE&query_params=%7B%220%22%3A%5B%220%22%5D%2C%2210%22%3A%22false%22%2C%2211%22%3A%22false%22%2C%2213%22%3A%22false%22%7D&format=json&device_id=****&api_key=256002347743983&sig=****&method=get&query_id=****
(18:34:34) http: destroying socket: 0x7f39349a90f0
(18:34:34) facebook: Reading 4 (flags: 0x2)
(18:34:34) facebook:   00000000  42 02 00 04                                       |B...|
(18:34:34) facebook:   00000010
(18:34:34) dnsquery: Performing DNS lookup for graph.facebook.com
(18:34:34) http: new socket created: 0x7f39343a26c0
(18:34:34) http: locking a (new) socket: 0x7f39343a26c0
(18:34:34) dns: Successfully sent DNS request to child 17879
(18:34:34) dns: Got response for 'graph.facebook.com'
(18:34:34) dnsquery: IP resolved for graph.facebook.com
(18:34:34) proxy: Attempting connection to 31.13.93.3
(18:34:34) proxy: Connecting to graph.facebook.com:443 with no proxy
(18:34:34) proxy: Connection in progress
(18:34:34) proxy: Connecting to graph.facebook.com:443.
(18:34:34) proxy: Connected to graph.facebook.com:443.
(18:34:34) nss: SSL version 3.3 using 128-bit AES-GCM with 128-bit AEAD MAC
Server Auth: 256-bit ECDSA, Key Exchange: 256-bit ECDHE, Compression: NULL
Cipher Suite Name: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
(18:34:34) nss: subject=CN=*.facebook.com,O="Facebook, Inc.",L=Menlo Park,ST=CA,C=US issuer=CN=DigiCert High Assurance CA-3,OU=www.digicert.com,O=DigiCert Inc,C=US
(18:34:34) nss: partial certificate chain
(18:34:34) certificate/x509/tls_cached: Starting verify for graph.facebook.com
(18:34:34) certificate/x509/tls_cached: Checking for cached cert...
(18:34:34) certificate/x509/tls_cached: ...Found cached cert
(18:34:34) nss/x509: Loading certificate from /home/michael/.purple/certificates/x509/tls_peers/graph.facebook.com
(18:34:34) certificate/x509/tls_cached: Peer cert matched cached
(18:34:34) nss/x509: Exporting certificate to /home/michael/.purple/certificates/x509/tls_peers/graph.facebook.com
(18:34:34) util: Writing file /home/michael/.purple/certificates/x509/tls_peers/graph.facebook.com
(18:34:34) nss: Trusting CN=*.facebook.com,O="Facebook, Inc.",L=Menlo Park,ST=CA,C=US
(18:34:34) certificate: Successfully verified certificate for graph.facebook.com
(18:34:34) http: Generated request headers:
POST /graphql HTTP/1.1
Host: graph.facebook.com
Connection: Keep-Alive
Accept: */*
Accept-Encoding: gzip, deflate
Content-Length: 325
Authorization: OAuth ****

(18:34:34) http: Got main header with code 200
(18:34:34) http: Got header: ETag: "****"
(18:34:34) http: Got header: fb-graphql-error-count: 0
(18:34:34) http: Got header: Content-Type: application/json; charset=UTF-8
(18:34:34) http: Got header: Facebook-API-Version: v1.0
(18:34:34) http: Got header: Expires: Sat, 01 Jan 2000 00:00:00 GMT
(18:34:34) http: Got header: Access-Control-Allow-Method: , OPTIONS
(18:34:34) http: Got header: Cache-Control: private, no-cache, no-store, must-revalidate
(18:34:34) http: Got header: Pragma: no-cache
(18:34:34) http: Got header: Vary: Origin
(18:34:34) http: Got header: X-FB-Rev: 1885582
(18:34:34) http: Got header: X-FB-Trace-ID: C0XuNxtQ3/c
(18:34:34) http: Got header: Access-Control-Allow-Origin: https://graph.facebook.com
(18:34:34) http: Got header: Access-Control-Expose-Headers: X-FB-Debug
(18:34:34) http: Got header: Access-Control-Allow-Credentials: true
(18:34:34) http: Got header: X-FB-Debug: 0051F2exgXbwbVO1a0J+YWqcwjyNQDe98cGvCvDZlknexelia//qrpFTFMxeOt7fLGNnFeJFENSfSDgMXtEj6w==
(18:34:34) http: Got header: Date: Sat, 15 Aug 2015 16:34:34 GMT
(18:34:34) http: Got header: Connection: keep-alive
(18:34:34) http: Got header: Content-Length: 10
(18:34:34) http: Got headers end
(18:34:34) http: Got response headers: ETag: "****"
fb-graphql-error-count: 0
Content-Type: application/json; charset=UTF-8
Facebook-API-Version: v1.0
Expires: Sat, 01 Jan 2000 00:00:00 GMT
Access-Control-Allow-Method: , OPTIONS
Cache-Control: private, no-cache, no-store, must-revalidate
Pragma: no-cache
Vary: Origin
X-FB-Rev: 1885582
X-FB-Trace-ID: C0XuNxtQ3/c
Access-Control-Allow-Origin: https://graph.facebook.com
Access-Control-Expose-Headers: X-FB-Debug
Access-Control-Allow-Credentials: true
X-FB-Debug: 0051F2exgXbwbVO1a0J+YWqcwjyNQDe98cGvCvDZlknexelia//qrpFTFMxeOt7fLGNnFeJFENSfSDgMXtEj6w==
Date: Sat, 15 Aug 2015 16:34:34 GMT
Connection: keep-alive
Content-Length: 10
(18:34:34) http: releasing a socket: 0x7f39343a26c0
(18:34:34) http: Request 0x7f3933599d10 performed successfully.
(18:34:34) facebook: HTTP Response (0x7f3933599d10):
(18:34:34) facebook:   Response Error: 200
(18:34:34) facebook:   Response Data: {"0":null}
(18:34:34) facebook: Parsing JSON: {"0":null}

(18:34:34) connection: Connection error on 0x7f3934379600 (reason: 0 description: Failed to parse thread information)
(18:34:34) http: destroying socket: 0x7f39343a26c0
(18:34:34) account: Disconnecting account **** (0x7f3933251e60)
(18:34:34) connection: Disconnecting connection 0x7f3934379600
(18:34:34) facebook: Writing 14 (flags: 0x0)
(18:34:34) facebook:   00000000  e0 00                                             |..|
(18:34:34) facebook:   00000010
(18:34:34) http: Cancelling all running HTTP connections
```

**Comment by jgeboski** [comment]
This has been resolved as of [cebf09840cbc](https://hg.pidgin.im/soc/2015/jgeboski/facebook/rev/cebf09840cbc), and will appear in the next release.

**Comment by jgeboski** [comment]
Also, thanks for the clean and detailed bug report.
